halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/DeploymentEnvironment.java

```diff
@@ -101,6 +103,7 @@ public static Size fromString(String name) {
   private String location;
   private CustomSizing customSizing = new CustomSizing();
   private GitConfig gitConfig = new GitConfig();
+  private List<String> haServices = new ArrayList<>();
```

Comment:
The halconfig's deploymentEnvironment field would look like this:
```
  deploymentEnvironment:
    size: SMALL
    type: Distributed
    accountName: k8s-v2
    updateVersions: true
    consul:
      enabled: false
    vault:
      enabled: false
    customSizing: {}
    gitConfig:
      upstreamUser: spinnaker
      originUser: joonlim
    haServices:
    - gate
    - clouddriver
```

Reply:
I would recommend switching this to either a list of structs (`name: gate`), or a key/value map `gate: ...` for the sake of allowing for future customization. We could definitely imagine exposing extra HA properties/config for these services in the future.